src/librustc_trans/back/link.rs

```diff
@@ -344,7 +350,7 @@ fn link_binary_output(sess: &Session,
     };
 
     match crate_type {
-        config::CrateTypeRlib => {
+        config::CrateTypeRlib | config::CrateTypeMetadata => {
```

Comment:
Hm so this is where I start to get a little uneasy, `link_rlib` has tons of logic for object files, native libraries, etc. Perhaps it'd be clearer and/or easier to just emit the metadata here to a dedicated file?